packages/datetime/src/dateInput.tsx

```diff
@@ -320,4 +321,36 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
     private setInputRef = (el: HTMLElement) => {
         this.inputRef = el;
     }
+
+    /**
+     * Translate a moment into a Date object, adjusting the moment timezone into the local one.
```

Comment:
where does the adjustment happen? i'm assuming it's _inside_ moment but it's not obvious _why_ this works.